File: src/main/java/org/kite9/diagram/dom/model/HasSVGRepresentation.java
package org.kite9.diagram.dom.model;

import org.kite9.diagram.model.DiagramElement;
import org.w3c.dom.Document;
import org.w3c.dom.Element;

/**
 * Means this {@link DiagramElement} can output XML
 * representing itself.
 * 
 * @author robmoffat
 *
 */
public interface HasSVGRepresentation extends DiagramElement {
	
	/**
	 * Creates SVG elements representing this DiagramElement and anything 
	 * nested within it.
	 */
	public Element output(Document d);

}

File: kite9-visualization/src/commonMain/kotlin/org/kite9/diagram/dom/model/HasSVGRepresentation.kt
package org.kite9.diagram.dom.model

import org.kite9.diagram.dom.processors.XMLProcessor
import org.kite9.diagram.model.DiagramElement
import org.w3c.dom.Document
import org.w3c.dom.Element

/**
 * Means this [DiagramElement] can output XML
 * representing itself.
 *
 * @author robmoffat
 */
interface HasSVGRepresentation : DiagramElement {

    /**
     * Creates SVG elements representing this DiagramElement and anything
     * nested within it.
     * @param p
     */
    fun output(d: Document, p: XMLProcessor): Element?
}